### PR TITLE
Add config file support, update yaml files with correct data for Compatibility Testing, fix generation support for queries(#6787)

### PR DESCRIPTION
**Pull request by GSharayu**
Compatibility tester needs that the yaml files used for testing (and the file paths within yaml files) reside in the resource directory path. We should change these so that the files can be absolute path. This will allow Pinot installations to specify their own test suite and run compat testing before upgrade to new pinot versions.

The tester script should take a directory name (absolute path) as the argument where all the yaml files are located. The yaml files should be named as pre-controller-upgrade.yaml, pre-broker-upgrade.yaml, etc (6 of them) . The shell script should be modified to pick up these yaml files from the directory specified.

In addition, the directory should contain a subdir called config under which all the config file names should reside (e.g. the table config files, schema files, kafka producer config files. etc.), possibly under subdirectories if the user chooses to put them that way. These config files are the ones referenced inside the yaml files.

In addition, also provide support to add config files for server, broker and controller. The file names are expected to be
1. BrokerConfig.conf
2. ServerConfig.conf
3. ControllerConfig.conf

This PR also fixes support different generation support queries

### CODE REVIEW

File: compatibility-verifier/compCheck.sh [hunk 1]
```diff
@@ -70,21 +70,31 @@ function waitForClusterReady() {
   sleep 2
   echo "Cluster ready."
 }
+
+#set config file is present or not
+function setConfigFile() {
```
Comment (mcvsubbu) [inline compatibility-verifier/compCheck.sh:75]:
```suggestion
function setConfigFileArg() {
```

File: compatibility-verifier/compCheck.sh [hunk 2]
```diff
@@ -70,21 +70,31 @@ function waitForClusterReady() {
...
+  if [[ -n $1 ]]; then
+    echo "-configFileName ${1}"
+  fi
+}
+
 # Given a component and directory, start that version of the specific component
 function startService() {
   serviceName=$1
   dirName=$2
+  local configFile=$(setConfigFile "$3")
```
Comment (mcvsubbu) [inline compatibility-verifier/compCheck.sh:85]:
```suggestion
  local configFileArg=$(setConfigFile "$3")
```

File: compatibility-verifier/compCheck.sh [hunk 3]
```diff
@@ -156,6 +170,13 @@ function absPath() {
   echo "$testSuiteDirPath"
 }
 
+#check if config file is present or not
+function checkConfigFile() {
+  if [[ -e $1 ]]; then
```
Comment (mcvsubbu) [inline compatibility-verifier/compCheck.sh:175]:
```suggestion
  if [[ -f $1 ]]; then
```

File: compatibility-verifier/compCheck.sh [hunk 4]
```diff
@@ -70,21 +70,31 @@ function waitForClusterReady() {
   sleep 2
   echo "Cluster ready."
 }
+
+#set config file is present or not
+function setConfigFile() {
+  if [[ -n $1 ]]; then
```
Comment (mcvsubbu) [inline compatibility-verifier/compCheck.sh:76]: why don't we just check if file exists (using -f) here and populate the configFileArg?

File: compatibility-verifier/compCheck.sh (hunk 2 above)
Comment (GSharayu) [reply compatibility-verifier/compCheck.sh]: done!

File: compatibility-verifier/compCheck.sh (hunk 1 above)
Comment (GSharayu) [reply compatibility-verifier/compCheck.sh]: done!

File: pinot-integration-tests/src/test/java/org/apache/pinot/compat/tests/QueryOp.java [hunk 5]
```diff
@@ -76,7 +76,11 @@ public void setExpectedResultsFileName(String expectedResultsFileName) {
   boolean runOp(int generationNumber) {
     System.out.println("Verifying queries in " + _queryFileName + " against results in " + _expectedResultsFileName);
     try {
-      return verifyQueries(generationNumber);
+      boolean queryVerificationResult = true;
+      for(int i = 1; i <= generationNumber; i++) {
+        queryVerificationResult &= verifyQueries(i);
```
Comment (mqliang) [inline pinot-integration-tests/src/test/java/org/apache/pinot/compat/tests/QueryOp.java:81]: Can early return here: 
```
for(int i = 1; i <= generationNumber; i++) {
        if (! verifyQueries(i) ) {
               return false;
        }
}
return true;
```

Comment (GSharayu) [reply pinot-integration-tests/src/test/java/org/apache/pinot/compat/tests/QueryOp.java]: done!

File: compatibility-verifier/compCheck.sh [hunk 6]
```diff
@@ -112,14 +122,18 @@ function stopService() {
 # Starts a Pinot cluster given a specific target directory
 function startServices() {
   dirName=$1
+  local controllerConfigFile=$2
```
Comment (mcvsubbu) [inline compatibility-verifier/compCheck.sh:125]: All config files are supposed to be in the config/ directory, right? So, we can just pass one arg here?

File: compatibility-verifier/compCheck.sh [hunk 7]
```diff
@@ -169,6 +183,11 @@ COMPAT_TESTER_PATH="pinot-integration-tests/target/pinot-integration-tests-pkg/b
 # create subdirectories for given commits
 workingDir=$1
 testSuiteDir=$(absPath "$2")
+
+brokerConfigFile=${testSuiteDir}/BrokerConfig.conf
```
Comment (mcvsubbu) [inline compatibility-verifier/compCheck.sh:187]:
```suggestion
brokerConfigFile=${testSuiteDir}/config/BrokerConfig.conf
```
Same for other files also. We want to have them all in the config/ directory

Comment (GSharayu) [reply compatibility-verifier/compCheck.sh]: done!

File: compatibility-verifier/compCheck.sh (hunk 6 above)
Comment (GSharayu) [reply compatibility-verifier/compCheck.sh]: passed config folder to startServices

File: compatibility-verifier/compCheck.sh [hunk 8]
```diff
@@ -169,6 +182,11 @@ COMPAT_TESTER_PATH="pinot-integration-tests/target/pinot-integration-tests-pkg/b
 # create subdirectories for given commits
 workingDir=$1
 testSuiteDir=$(absPath "$2")
+
+brokerConfigFile=${testSuiteDir}/config/BrokerConfig.conf
```
Comment (mcvsubbu) [inline compatibility-verifier/compCheck.sh:186]: Can you set up front with `BROKER_CONF="BrokerConfig.conf"` and then use `${BROKER_CONF}` everywhere? Similarly the other two?

Thanks.

**Comment by mqliang** [comment]
LGTM

File: compatibility-verifier/compCheck.sh [hunk 9]
```diff
@@ -112,14 +121,18 @@ function stopService() {
 # Starts a Pinot cluster given a specific target directory
 function startServices() {
   dirName=$1
+  local CONTROLLER_CONF=$2/ControllerConfig.conf
```
Comment (mcvsubbu) [inline compatibility-verifier/compCheck.sh:124]: Not local, move this up in the script to make it non-local (remove the local keyword). You can put it in line 42

File: compatibility-verifier/compCheck.sh [hunk 10]
```diff
@@ -169,6 +182,11 @@ COMPAT_TESTER_PATH="pinot-integration-tests/target/pinot-integration-tests-pkg/b
 # create subdirectories for given commits
 workingDir=$1
 testSuiteDir=$(absPath "$2")
+
+BROKER_CONF=${testSuiteDir}/config/BrokerConfig.conf
```
Comment (mcvsubbu) [inline compatibility-verifier/compCheck.sh:186]: You will now be able to reuse ${BROKER_CONF} everywhere in the script.

File: compatibility-verifier/compCheck.sh (hunk 9 above)
Comment (GSharayu) [reply compatibility-verifier/compCheck.sh]: updated

File: compatibility-verifier/compCheck.sh (hunk 10 above)
Comment (GSharayu) [reply compatibility-verifier/compCheck.sh:183]: done!

**Review by mcvsubbu — APPROVED** [review]

**Pull request merged**